packages/host-arch-riscv64/host-arch-riscv64.1/opam

```diff
@@ -0,0 +1,16 @@
+opam-version: "2.0"
+synopsis: "OCaml on 64-bit RISC-V"
+description: """
+This package is installed if the underlying OCaml compiler is for 64-bit RISC-V.
+
+Precisely, this means `ocamlopt -config-var architecture` equals `riscv`.
+
+This package may be used in depends or conflicts fields of dependent packages
+to indicate either a requirement or an incompatibility with this
+architecture."""
+maintainer: "David Allsopp <david@tarides.com>"
+authors: "David Allsopp"
+license: "CC0-1.0+"
+homepage: "https://opam.ocaml.org"
+bug-reports: "https://github.com/ocaml/opam-repository/issues"
+conflict-class: "ocaml-host-architecture"
```

Comment:
```suggestion
conflict-class: "host-arch"
```

to match the now-reserved prefix for package names

needs to be applied to other packages too

Reply:
actually maybe using just the prefix is a bit short, especially for some other packages like `arch-`. So possibly we should have `meta-package-<prefix>` or `opam-setup-selection-<prefix>` or something like that.

We'll discuss this during the meeting (or anyone should feel free to drop comments here) and decide on something.

Reply:
What was the conclusion for these ones? The PR at the moment introduces:
- `msys2-environment`
- `ocaml-architecture-setting` & `ocaml-system-setting`
- `ocaml-host-architecture` & `ocaml-host-system`
- `ocaml-mingw-env` & `ocaml-msvc-env`

Prior to this PR, opam-repository's only use of `conflict-class` is `ocaml-core-compiler` (which is why the feature was added)

Reply:
I want to strive for regularity so I think the conflict classes should be as follows:

- `msys2-env` (consistent use of `env` with other classes)
- `ocaml-arch` (consistent use of `arch` as per package name, `setting` doesn't appear in package name) & `ocaml-system`
- `ocaml-host-arch` & `ocaml-host-system`
- `ocaml-mingw-env` & `ocaml-msvc-env`

Moreover, would it make sense for the `ocaml-{mingw,msvc}{32,64}-env` to be named `ocaml-{mingw,msvc}-env-{32,64}` instead? The arch packages have the prefix set and then afterwards the variant. Whereas the mingw/msvc env packages have the variant in the middle of the prefix.

Reply:
`conflict-class`s all updated. I like the switch to `ocaml-env-...` but I've stuck with the existing set of `mingw32`, `mingw64`, `msvc32` and `msvc64` principally because these serve the same purpose as the MSYS2 Environments. i.e. `ocaml-env-mingw64` is performing the Cygwin-related function that `msys2-mingw64` performs.

Reply:
the conflict classes are good as they currently stand, thanks